tests/modules/entity_module_test/src/Entity/EntityWithRevisionRoutes.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\entity_module_test\Entity\EntityWithRevisionRoutes.
+ */
+
+namespace Drupal\entity_module_test\Entity;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\entity\EntityKeysFieldsTrait;
+use Drupal\entity\Revision\EntityRevisionLogTrait;
+
+/**
+ * @ContentEntityType(
+ *   id = "entity_test__rev_routes",
+ *   label = @Translation("Entity test with revision routes"),
+ *   handlers = {
+ *     "storage" = "\Drupal\Core\Entity\Sql\SqlContentEntityStorage",
+ *     "route_provider" = {
+ *       "revision" = "\Drupal\entity\Routing\RevisionRouteProvider",
+ *     },
+ *   },
+ *   base_table = "entity_test__revision_routes",
+ *   data_table = "entity_test__revision_routes__field_data",
+ *   revision_table = "entity_test__revision_routes__revision",
+ *   revision_data_table = "entity_test__revision_routes__field_revision",
+ *   translatable = TRUE,
+ *   revisionable = TRUE,
+ *   admin_permission = "administer entity_test__revision_routes",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "revision" = "vid",
+ *     "langcode" = "langcode",
+ *   },
+ *   links = {
+ *     "revision" = "/entity_test__rev_routes/{entity_test__rev_routes}/revisions/{entity_test__rev_routes_revision}/view",
+ *   }
+ * )
+ */
+class EntityWithRevisionRoutes extends ContentEntityBase {
```

Comment:
As discussed, I think it would be good to have a single test entity that does all the things. Servers as an example and makes tests faster as we have fewer tables to create.